optimum/pipelines.py

```diff
@@ -0,0 +1,104 @@
+from sre_constants import ANY
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Tuple, Union
+
+from transformers import (
+    AutoTokenizer,
+    FeatureExtractionPipeline,
+    Pipeline,
+    PreTrainedTokenizer,
+    QuestionAnsweringPipeline,
+    TextClassificationPipeline,
+    TokenClassificationPipeline,
+    ZeroShotClassificationPipeline,
+)
+from transformers import pipeline as transformers_pipeline
+from transformers.feature_extraction_utils import PreTrainedFeatureExtractor
+from transformers.pipelines import TASK_ALIASES
+
+from optimum.utils import is_onnxruntime_available
+
+
+SUPPORTED_TASKS = {}
+
+if is_onnxruntime_available():
+    from optimum.onnxruntime import (
+        OnnxForFeatureExtraction,
+        OnnxForQuestionAnswering,
+        OnnxForSequenceClassification,
+        OnnxForTokenClassification,
+        OnnxModel,
+    )
+
+    SUPPORTED_TASKS = {
+        "feature-extraction": {
+            "impl": FeatureExtractionPipeline,
+            "class": (OnnxForFeatureExtraction,) if is_onnxruntime_available() else (),
+            "default": "distilbert-base-cased",
+        },
+        "text-classification": {
+            "impl": TextClassificationPipeline,
+            "class": (OnnxForSequenceClassification,) if is_onnxruntime_available() else (),
+            "default": "distilbert-base-uncased-finetuned-sst-2-english",
+        },
+        "token-classification": {
+            "impl": TokenClassificationPipeline,
+            "class": (OnnxForTokenClassification,) if is_onnxruntime_available() else (),
+            "default": "dbmdz/bert-large-cased-finetuned-conll03-english",
+        },
+        "question-answering": {
+            "impl": QuestionAnsweringPipeline,
+            "class": (OnnxForQuestionAnswering,) if is_onnxruntime_available() else (),
+            "default": "distilbert-base-cased-distilled-squad",
+        },
+        "zero-shot-classification": {
+            "impl": ZeroShotClassificationPipeline,
+            "class": (OnnxForSequenceClassification,) if is_onnxruntime_available() else (),
+            "default": "facebook/bart-large-mnli",
+        },
+    }
+
+
+def optimum_pipeline(
```

Comment:
What about `onnxruntime_pipeline`? I am suggesting this because we might extend `OptimizedModel` for other inference engines and we could imagine having other pipelines in optimum.

Reply:
If possible, I think it would be nice if we can have a single pipeline function, similar to how `transformers` just has `pipeline()` that works for both PyTorch & TensorFlow backends.

Reply:
I created the `optimum_pipeline` wrapper to have a check on which task is supported and have a default model. Not included in the code but in the [documentation](https://moon-ci-docs.huggingface.co/docs/optimum/pr_113/en/pipelines#optimum-pipeline-usage) is what @lewtun said instead of `framework` i thought about `accelerator`. 

```python
from optimum import optimum_pipeline

classifier = optimum_pipeline(task="text-classification", accelerator="onnx")
```